### DISCUSSION TITLE
When attributes from Prototype are applied when an object is created?

### DISCUSSION THREAD

**Discussion by Chetral** [evennia/evennia#3587] [Evennia Questions] [answered]
I'm dealing with this problem:
I have a prototype, at some points i added a `"wear":{"slot":["obj_prototype_name", material_id], ....}`

A function in the mob typeclass read this attribute, spawn the given object (of the material required) and make the mob wearing it in the given slot.

My problem is: if i call the func from "at_object_creation" the self.wear is empty. 
If i run it from `py` after creation it works.

I also tried putting it in the "EquimentHandler" that is a @property of mob typeclasse:
```py
@property
def eq(self):
    return EquipmentHandler
```
    
In this case it works, but fires only the first time i LOOK at the mob.
The latter solution works, but is just a workaround. Do exist a default "post_creation" hook on DefaultObjects?

**Comment by lundexurkai** [comment]
`at_object_creation` is only called **once** the object is created. And from there the attributes of the prototype are set. So it makes sense that the `wear` would be empty when `at_object_creation` is called.

**Comment by InspectorCaracal** [comment]
The question of "where do I run initial creation logic that happens *after* the prototype is applied?" is unfortunately not a straight forward one to answer. Here's the relevant, still open feature request: https://github.com/evennia/evennia/issues/3095

**Reply by Chetral** [reply]
For now, my homemade solution is a "post_object_creation" hook that is called from the spawn command.
That because i already had a modified "CmdSpawn" to place objects in the wilderness. But still is not a correct solution, as if i create objects from code (for example with the dungeon contrib in EvAdventure) i need to call the new hook there too.

**Reply by InspectorCaracal** [reply, marked as answer]
Yeah, exactly. As is mentioned in the feature issue discussion, there likely needs to be both an `at_post_spawn` (or just `at_spawn`) hook to called specifically after being spawned, and an `at_post_object_creation` hook to be called at the very end of the object creation process for all objects.

**Reply by Chetral** [reply]
I marked this as answer, as it recall to the issue that is taking care of the question.
